File: app/src/main/java/com/example/studyanimation/opengl/surfacetexture/MyGLSurfaceView.java
package com.example.studyanimation.opengl.surfacetexture;

import android.content.Context;
import android.opengl.GLSurfaceView;

/**
 * Created by palnak on 2/8/2017.
 */

public class MyGLSurfaceView extends GLSurfaceView {

    MyGL20Renderer renderer;
    public MyGLSurfaceView(Context context)
    {
        super(context);

        setEGLContextClientVersion(2);
        setEGLConfigChooser(8,8,8,8,16,0);
        setZOrderOnTop(true);


        renderer = new MyGL20Renderer((MainCameraGLActivity)context);

        setRenderer(renderer);

        setRenderMode(GLSurfaceView.RENDERMODE_WHEN_DIRTY);//means "do not call onDrawFrame() unless something explicitly requests rendering with requestRender()

    }

    public MyGL20Renderer getRenderer()
    {
        return renderer;
    }
}

File: app/src/main/java/com/example/studyanimation/image_graphic/opengl/MyGLSurfaceView.kt
package com.example.studyanimation.image_graphic.opengl

import android.content.Context
import android.opengl.GLSurfaceView
import android.view.MotionEvent

class MyGLSurfaceView(context: Context) : GLSurfaceView(context) {

    private val renderer: MyGLRenderer

    private var previousX: Float = 0f
    private var previousY: Float = 0f

    init {

        // Create an OpenGL ES 2.0 context
        setEGLContextClientVersion(2)

        renderer = MyGLRenderer()

        // Set the Renderer for drawing on the GLSurfaceView
        setRenderer(renderer)


        //그리기 데이터가 변경될 때만 뷰를 그리도록 렌더링 모드를 설정하는 것입니다.
        renderMode = RENDERMODE_WHEN_DIRTY
        //이처럼 설정하면 requestRender()를 호출할 때까지 GLSurfaceView 프레임을 다시 그리지 않습니다. 그러면 이 샘플 앱에 더 효율적입니다.
    }

    override fun onTouchEvent(e: MotionEvent): Boolean {
        // MotionEvent reports input details from the touch screen
        // and other input controls. In this case, you are only
        // interested in events where the touch position changed.

        val x: Float = e.x
        val y: Float = e.y

        when (e.action) {
            MotionEvent.ACTION_MOVE -> {

                var dx: Float = x - previousX
                var dy: Float = y - previousY

                // reverse direction of rotation above the mid-line
                if (y > height / 2) {
                    dx *= -1
                }

                // reverse direction of rotation to left of the mid-line
                if (x < width / 2) {
                    dy *= -1
                }

                renderer.angle += (dx + dy) * TOUCH_SCALE_FACTOR
                requestRender()
            }
        }

        previousX = x
        previousY = y
        return true
    }

    companion object {
        private const val TOUCH_SCALE_FACTOR: Float = 180.0f / 320f
    }


}
